h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -1077,9 +1077,16 @@ static void closeNewChunks(NewChunk[] nchks) {
     fs.blockForPending();
   }
 
+  // Build real Vecs from loose Chunks, and finalize this Frame.  Called once
+  // after any number of [create,close]NewChunks. This method also unlocks the frame.
+  void finalizePartialFrame(long[] espc, String[][] domains, byte[] types) {
+    finalizePartialFrameAndLeaveLocked(espc, domains, types);
+    unlock();
+  }
+  
   // Build real Vecs from loose Chunks, and finalize this Frame.  Called once
   // after any number of [create,close]NewChunks.
-  void finalizePartialFrame( long[] espc, String[][] domains, byte[] types ) {
+  void finalizePartialFrameAndLeaveLocked(long[] espc, String[][] domains, byte[] types) {
```

Comment:
Would be a lot nicer to rename this method (if this can be done) to `finalizePartialFrame` and introduce a new argument `boolean lock` - this way, the method would be generic enough. 

I mean, if we can't change the signature of this method, we might always leave it as-is, introduce a new one with the parameter and let the old one call the new one with `true` value which means "lock it, please".

Reply:
ok np, will do. I don't have any preferences.